Host aggregate cleanup - UTC time fix and  better logging

AnishReddyRavula opened the pull request:
- Use UTC time as blazar DB uses UTC timestamps
- This caused an issue with the monitor plugin moving to `freepool` from a current active reservation as the `get_most_recent_reservation_info_by_host_id` always picks an old reservation that is in deleted or error state
- Better logging with warning messages and debug messages
- Skip a host cleanup if the blazar picked up aggregate is different from the current aggregate in nova - give a warning

AnishReddyRavula committed:
Use UTC time as blazar DB uses utc timestamps

This caused an issue with monitor plugin moving to freepool from a current active reservation as the get_most_recent_reservation_info_by_host_id always picks an old reservation which is in deleted or error state

AnishReddyRavula committed:
Better logging with warning messages and debug messages

Skip a host cleanup if the blazar picked up aggregate is different from current aggregate in nova - give a warning

Mark-Powers commented on blazar/plugins/oshosts/host_plugin.py:
```diff
@@ -987,28 +987,41 @@ def poll_resource_failures(self):
+                if curr_agg.name == freepool.name:
                     # if the host is already in freepool skip it
-                    LOG.info(f"{host['hypervisor_hostname']} is already in a freepool - skipping it")
+                    LOG.debug(f"{host['hypervisor_hostname']} is already in a freepool - skipping aggregate clean up")
```
Do we need this case? Isn't this just the correct behavior?

AnishReddyRavula replied:
We would like to do nothing if the host is already in a freepool.

msherman64 replied:
this is here because we depend on the `continue` for these cases, which I admit is a bit confusing

AnishReddyRavula committed:
Dry run config opt for polling monitor

host plugin monitor to not move hosts between aggregates when dry_polling_monitor is enabled

Remove the condition to not move hosts if the most recent reservation is not equal to the current aggregate.
There might be a case when a host is not in any reservation or in any aggregate, so doing this will not help

Testcase to use the dry_polling_monitor flag

msherman64 reviewed (approved):
LGTM
